src/main/java/com/rultor/agents/github/qtn/ReleaseTag.java

```diff
@@ -0,0 +1,127 @@
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.rultor.agents.github.qtn;
+
+import com.jcabi.aspects.Immutable;
+import com.jcabi.github.Release;
+import com.jcabi.github.Repo;
+import com.jcabi.github.Smarts;
+import java.io.IOException;
+import java.util.regex.Pattern;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+
+/**
+ * Release Tag validator, ensures not releasing already outdated tags.
+ *
+ * @author Armin Braun (me@obrown.io)
+ * @version $Id$
+ * @since 1.62
+ */
+@Immutable
+final class ReleaseTag {
+
+    /**
+     * Pattern matching semantically valid versions.
+     */
+    private static final Pattern VERSION_PATTERN =
+        Pattern.compile("^(\\d+\\.)*(\\d+)$");
+
+    /**
+     * Github.
+     */
+    private final transient Repo repo;
+
+    /**
+     * Tag-name of the release.
+     */
+    private final transient String name;
+
+    /**
+     * Ctor.
+     * @param rpo Github repo
+     * @param version String release tag name
+     */
+    ReleaseTag(final Repo rpo, final String version) {
+        this.repo = rpo;
+        this.name = version;
+    }
+
+    /**
+     * Checks if this tag can be released.
+     * A tag can be released if it is either not named as a semantically
+     * correct version or has a higher version number than all existing tags.
+     * @return Boolean
+     * @throws IOException on error.
+     */
+    public boolean allowed() throws IOException {
+        return !ReleaseTag.valid(this.name)
+            || ReleaseTag.newer(this.reference(), this.name);
+    }
+
+    /**
+     * Returns the tag name of the highest version from the repo.
+     * @return String name of the highest released version.
+     * @throws IOException on error.
+     */
+    public String reference() throws IOException {
+        String tag = "0";
+        final Iterable<Release.Smart> rels =
+            new Smarts<>(this.repo.releases().iterate());
+        for (final Release.Smart rel : rels) {
+            final String version = rel.tag();
+            if (ReleaseTag.valid(version) && ReleaseTag.newer(tag, version)) {
+                tag = version;
+            }
+        }
+        return tag;
+    }
+
+    /**
+     * Checks that a tag is newer than a given reference.
+     * @param reference String
+     * @param tag String
+     * @return Boolean true if tag is new than reference.
```

Comment:
@original-brownbear why not just `True if tag is new than reference` without dot?